From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 11 Apr 2017 15:17:48 +0200
Subject: [PATCH] Don't close fds when executing programs

Tools that execute meson might open file descriptors for the programs to
use. Such as a file descriptor for a logfile.
---
 mesonbuild/mesonlib.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mesonbuild/mesonlib.py b/mesonbuild/mesonlib.py
index c7368d577ad8..7f328f5e3334 100644
--- a/mesonbuild/mesonlib.py
+++ b/mesonbuild/mesonlib.py
@@ -492,6 +492,7 @@ def expand_arguments(args):
 
 def Popen_safe(args, write=None, stderr=subprocess.PIPE, **kwargs):
     p = subprocess.Popen(args, universal_newlines=True,
+                         close_fds=False,
                          stdout=subprocess.PIPE,
                          stderr=stderr, **kwargs)
     o, e = p.communicate(write)
